src/main/java/core/basesyntax/DataConverterImpl.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    private static final int MAX_NUMBER_OF_PARTS = 3;
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        for (String line : data) {
+            if (line.trim().isEmpty() || line.startsWith("type")) {
+                continue;
+            }
+            String[] parts = line.split(",");
+            if (parts.length != MAX_NUMBER_OF_PARTS) {
+                throw new RuntimeException("Invalid line format" + line);
+            }
+            String typeCode = parts[0];
+            String fruit = parts[1];
+            int quantity;
+            try {
+                quantity = Integer.parseInt(parts[2]);
+            } catch (NumberFormatException e) {
+                throw new RuntimeException("Invalid quanity: " + parts[2], e);

```

Comment:
The exception message should have a space after the colon for better readability. It should be 'Invalid quantity: ' instead of 'Invalid quanity:'